src/Microsoft.Azure.ServiceBus/Filters/FilterExtensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿namespace Microsoft.Azure.ServiceBus
+{
+    using System.Xml.Linq;
+    using Microsoft.Azure.ServiceBus.Management;
+
+    internal static class FilterExtensions
+    {
+        public static Filter ParseFromXElement(XElement xElement)
+        {
+            var attribute = xElement.Attribute(XName.Get("type", ManagementClientConstants.XmlSchemaNs));
+            if (attribute == null)
+            {
+                return null;
+            }
+
+            switch (attribute.Value)
+            {
+                case "SqlFilter":
+                    return SqlFilterExtensions.ParseFromXElement(xElement);
+                case "CorrelationFilter":
+                    return CorrelationFilterExtensions.ParseFromXElement(xElement);
+                case "TrueFilter":
+                    return new TrueFilter();
+                case "FalseFilter":
+                    return new FalseFilter();
+                default:
+                    return null;
+            }
+        }
+
+        public static XElement Serialize(this Filter filter)
+        {
```

Comment:
FalseFilter and TrueFilter have been taken off. Merge issues?

Reply:
Those two will fall under `SqlFilter` and that `Serialize` will be invoked as they both inherit from `SqlFilter`.

Reply:
In that case, should handle it differently. Will update in a bit, leave this as TODO for now.